sdk/core/core/src/az_json_string.c

```diff
@@ -13,10 +13,10 @@ AZ_NODISCARD AZ_INLINE az_result_byte az_hex_to_digit(az_result_byte const c) {
     return c - '0';
   }
   if ('a' <= c && c <= 'f') {
-    return c - AZ_HEX_LOWER_OFFSET;
```

Comment:
again, just use a builtin function here

Reply:
I've tried but it fails in VS if `c` is less than -1.